### PR TITLE
[14.0][MIG] rental_pricelist: Migration to 14.0

### PR DESCRIPTION
**Pull request by chandni299**

### CODE REVIEW

File: rental_pricelist/models/product.py [hunk 1]
```diff
@@ -342,39 +353,45 @@ def write(self, vals):
...
+                ] = res.income_analytic_account_id.id
+            if "expense_analytic_account_id" in vals:
+                ext_vals["expense_analytic_account_id"] = vals.get(
+                    "expense_analytic_account_id", False
+                )
+            else:
+                ext_vals[
+                    "expense_analytic_account_id"
+                ] = res.expense_analytic_account_id.id
         res.write(ext_vals)
```
Comment (yweng8111) [inline rental_pricelist/models/product.py:379]: ext_vals contains only the value of one created product. It should not be written in all the new created products.

File: rental_pricelist/tests/test_rental_pricelist.py [hunk 2]
```diff
@@ -177,13 +186,13 @@ def test_01_rental_onchange_productA(self):
...
         line.onchange_display_product_id()
-        res = line.product_id_change()
+        line.product_id_change()
         # check uom domain
-        check_uom_domain = False
-        if "domain" in res and "product_uom" in res["domain"]:
-            self.assertEqual(len(res["domain"]["product_uom"][0][2]), 3)
-            check_uom_domain = True
-        self.assertTrue(check_uom_domain)
+        # check_uom_domain = False
```
Comment (yweng8111) [inline rental_pricelist/tests/test_rental_pricelist.py]: Why do you deactivate this check?

File: rental_pricelist/tests/test_rental_pricelist.py [hunk 3]
```diff
@@ -237,13 +246,13 @@ def test_02_rental_onchange_productC(self):
...
         line.onchange_display_product_id()
-        res = line.product_id_change()
+        line.product_id_change()
         # check uom domain
-        check_uom_domain = False
-        if "domain" in res and "product_uom" in res["domain"]:
-            self.assertEqual(res["domain"]["product_uom"][0][2][0], self.uom_month.id)
-            check_uom_domain = True
-        self.assertTrue(check_uom_domain)
+        # check_uom_domain = False
```
Comment (yweng8111) [inline rental_pricelist/tests/test_rental_pricelist.py]: Why do you deactivate this check?

File: rental_pricelist/views/product_view.xml [hunk 4]
```diff
@@ -56,7 +56,6 @@
                                         />
                                         <field name="compute_price" invisible="1" />
                                         <field name="applied_on" invisible="1" />
-                                        <field name="product_id" invisible="1" />
```
Comment (yweng8111) [inline rental_pricelist/views/product_view.xml]: Why do you delete this line?

File: rental_pricelist/views/product_view.xml [hunk 5]
```diff
@@ -101,7 +100,6 @@
...
```
Comment (yweng8111) [inline rental_pricelist/views/product_view.xml]: Why do you delete this line?

File: rental_pricelist/views/product_view.xml [hunk 6]
```diff
@@ -146,7 +144,6 @@
...
```
Comment (yweng8111) [inline rental_pricelist/views/product_view.xml]: Why do you delete this line?

File: rental_pricelist/tests/test_rental_pricelist.py (hunk 2 above)
Comment (chandni299) [reply rental_pricelist/tests/test_rental_pricelist.py]: in v14 `def product_id_change()` in model `sale.order.line`, we do not have "domain" as key in retrun `res`

File: rental_pricelist/views/product_view.xml (hunk 4 above)
Comment (chandni299) [reply rental_pricelist/views/product_view.xml]: in v14 due to view changes, either we need to remove it or add more fields to adjust our field `product_id`

File: rental_pricelist/models/sale.py [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+        if self.display_product_id.rental_of_day:
+            uom_ids.append(time_uoms["day"].id)
+        if self.display_product_id.rental_of_hour:
+            uom_ids.append(time_uoms["hour"].id)
+        return uom_ids
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        res = super(SaleOrderLine, self).product_id_change()
+        if self.rental and "domain" in res and "product_uom" in res["domain"]:
```
Comment (yweng8111) [inline rental_pricelist/models/sale.py]: So We should adjust this if clause and add the domain, if res doesn't contain "domain".

File: rental_pricelist/tests/test_rental_pricelist.py (hunk 2 above)
Comment (yweng8111) [reply rental_pricelist/tests/test_rental_pricelist.py]: Then we can adjust the if clause in product_id_change() and add the key domain.

**Comment by chandni299** [comment]
@yweng8111 you can check the correction.

**Review by yweng8111 — APPROVED** [review]

**Review by marylla — APPROVED** [review]
I made functional review as described in usage section of manifest.
It works as expected.

**Comment by rousseldenis** [comment]
@marylla Could you use ocabot to merge ? Thanks

**Pull request merged**